en/components/general/how-to/how-to-customize-theme.md

```diff
@@ -0,0 +1,329 @@
+
+```scss
+/* You can add global styles to this file, and also import other style files */
+
+// Standard CSS normalize, comment out if not required or using a different module
+@use "minireset.css/minireset";
+@use "@infragistics/igniteui-angular/theming" as *;
+
+@include core();
+@include typography();
+@include light-theme($light-material-palette);
+
+body {
+  background: hsla(var(--ig-surface-500));
+  color: var(--ig-surface-500-contrast);
+}
+
+html, body {
+  height: 100%;
+}
+
+.igx-typography {
+  h1, h2, h3, h4, h5, h6, p, .igx-typography__body-1 {
+    margin: 0;
+  }
+}
+
+.outer-wrapper > *:not(router-outlet) {
+  width: 100%;
+}
+```
+
+## Theme Customization
+
+We want a dark variant of the same theme, add our own [color palette](../../themes/palettes.md) to match our branding, and change the font to `Poppins` instead of the default `Titillium Web`, all of which we can change directly through the App Builder and we can push the change as a pull request from the App Builder into the repository.
+
+<img class="responsive-img"  src="../../../images/general/theming-walkthrough/getting-started-appbuilder-theming.png" />
+
+The updated `styles.scss` looks like this:
+
+```scss
+@include core();
+@include typography($font-family: "Poppins");
+
+$custom-palette: palette(
+  $primary: #1028c7,
+  $secondary: #e0d94c,
+  $surface: #000000,
+  $gray: #FFFFFF);
+@include theme($palette: $custom-palette, $schema: $dark-material-schema);
+```
+
+As you can see, the code generation changed from the specific `@include light-theme($light-material-palette);`, which is the [default theme](../../themes/sass/presets/material.md) and [color palette](../../themes/palettes.md), to a generic [`theme()`]({environment:sassApiUrl}/index.html#mixin-theme) include, which provides as parameters our custom color palette and a [dark material schema](../../themes/sass/schemas.md) for the theming structure. The running Angular app result looks like this now:
+
+<img class="responsive-img"  src="../../../images/general/theming-walkthrough/getting-started-dark-app.png" />
+
+We want to dig deeper and customize a specific [component theme](../../themes/sass/component-themes.md) in our application and we will do this by bringing in the CSS variables for an individual component theme, in this case the grid toolbar theme. 
+
+```scss
+@include core();
+@include typography($font-family: "Poppins");
+
+$primary: #1028c7;
+
+/* All of the components will use this custom color palette */
+$custom-palette: palette(
+  $primary: $primary,
+  $secondary: #e0d94c,
+  $surface: #000000,
+  $gray: #FFFFFF);
+@include theme($palette: $custom-palette, $schema: $dark-material-schema);
+
+/* Grid Toolbar */
+/* All grid toolbars will have custom background and elevations */
+$toolbar-theme: grid-toolbar-theme(
+  $background-color: $primary,
+  $elevations: 0px 15px 30px 0px rgba(0,0,0,0.4)
+);
```

Comment:
```suggestion
$toolbar-theme: grid-toolbar-theme(
  $background-color: $primary
);
```

Remove this as it does nothing. We expect it to be a map of elevations and not a single box-shadow. The map must conform to the existing material elevations map. Furthermore, it seems, there's currently a bug with toolbar theme as it does nothing with the `$elevations` property, even if it's a valid elevations map.